### PR TITLE
docs: added CollectionOfDummy to dummies.md

### LINKED ISSUE #675: Helper method for list of dummies
I really like the `Dummy` feature of FakeItEasy. One thing that I find myself needing to do on every project is to create a collection of dummies (e.g. for seeding a test database).

I always end up writing an extension method that give me a fluent syntax for this, like:
```cs
IEnumerable<User> users = 42.Dummy<User>();
```
the code for which is just something like:
```cs
public static class DummyHelpers
{
    public static IEnumerable<T> Dummy<T>(this int count)
    {
        for (var i = 0; i < count; i++)
        {
            yield return A.Dummy<T>();
        }
    }
}
```

Would you consider adding something like this to the framework? If so I would be happy to write up a pull request with any necessary tests.

---

- [x] implement - ~~#702~~
- [x] approve API change - ~~#705~~
- [ ] update docs - #710
- [x] update https://github.com/FakeItEasy/FakeItEasy/wiki/Changes-in-version-2.0

### PR DESCRIPTION
**Pull request by adamralph**
closes #675

### CODE REVIEW

File: docs/dummies.md [hunk 1]
```diff
@@ -63,7 +71,7 @@ of approaches in turn, until one succeeds:
   Fake `T`.
 1. If `T` is a value type, the Dummy will be a `T` created via
   `Activator.CreateInstance`.
-1. If nothing above matched, then `T` is a class. Loop over all its constructors in _descending order of argument list length_.  
+1. If nothing above matched, then `T` is a class. Loop over all its constructors in _descending order of argument list length_.
```
Comment (adamralph) [inline docs/dummies.md]: EditorConfig :wink:

**Comment by thomaslevesque** [comment]
LGTM :+1: 

Unrelated to this PR, but shouldn't the method be named `CollectionOfDummies` instead of `CollectionOfDummy` (and similarly, `CollectionOfFakes` instead of `CollectionOfFake`)? Sure, you could make a collection with only one dummy or fake, but what's the point?

**Comment by adamralph** [comment]
@thomaslevesque I believe the intention behind the singular "fake" in `A.CollectionFake<T>()` is so that, e.g.

```C#
A.CollectionOfFake<Widget>();
```

reads fluently, i.e. "a collection of fake widget". (Of course, it would be better if "widget" was pluralised.)

This tends to read better than

```C#
A.CollectionOfFakes<Widget>();
```

i.e. "a collection of fakes widget".

`A.CollectionOfDummy<T>()` followed this precedent.

**Comment by thomaslevesque** [comment]
You're right, it does read more fluently with the type argument.

**Pull request merged**